runhouse/rns/run.py

```diff
@@ -0,0 +1,560 @@
+import copy
+import json
+import logging
+import sys
+from datetime import datetime
+from io import StringIO
+from typing import Any, Callable, List, Optional, Union
+
+from ray import cloudpickle as pickle
+
+from runhouse import blob
+from runhouse.rh_config import obj_store, rns_client
+from runhouse.rns.api_utils.utils import log_timestamp
+from runhouse.rns.folders.folder import Folder, folder
+from runhouse.rns.function import Function
+from runhouse.rns.hardware import Cluster
+from runhouse.rns.resource import Resource
+from runhouse.rns.top_level_rns_fns import resolve_rns_path
+
+logger = logging.getLogger(__name__)
+
+
+class Run(Resource):
+    RESOURCE_TYPE = "run"
+
+    LOCAL_RUN_PATH = f"{rns_client.rh_directory}/runs"
+    RUN_CONFIG_FILE = "config_for_rns.json"
+
+    NOT_STARTED_STATUS = "NOT_STARTED"
+    RUNNING_STATUS = "RUNNING"
+    COMPLETED_STATUS = "COMPLETED"
+    ERROR_STATUS = "ERROR"
+
+    # Supported Run types
+    CLI_RUN = "CLI"
```

Comment:
would `CMD_RUN = "CMD"` make a little more sense, since we support both run_python and run, and also the differentiating factor is the `cmds` param passed into the function